src/WebJobs.Script/Host/FunctionMetadataManager.cs

```diff
@@ -285,5 +293,55 @@ private void AddMetadataFromCustomProviders(IEnumerable<IFunctionProvider> funct
                 }
             }
         }
+
+        private void ValidateHostJsonIfNoFunctionsFound()
+        {
+            string hostFilePath = Path.Combine(_scriptOptions.Value.RootScriptPath, ScriptConstants.HostMetadataFileName);
+
+            // Search for the host.json file within nested directories to verify scenarios where it isn't located at the root. This situation often occurs when a function app has been improperly zipped.
+            IEnumerable<string> hostJsonFiles = null;
+            try
+            {
+                hostJsonFiles = Directory.GetFiles(_scriptOptions.Value.RootScriptPath, ScriptConstants.HostMetadataFileName, SearchOption.AllDirectories)
+                    .Where(file => !file.Equals(hostFilePath, StringComparison.OrdinalIgnoreCase));
+            }
+            catch
+            {
+                // Ignore any exceptions while looking for host.json files in nested directories.
+            }
+
+            if (IsDefaultHostConfig())
```

Comment:
My understanding is that the field `defaultHostConfig` is being used to differentiate between the host.json that comes with Customer payload versus the Host.json created by Host if it does not find the file in Cx payload. Is this correct?

If Customer has an app with no functions and their host.json also has this field, we may end up printing incorrect log?